src/probes/klog_main.c

```diff
@@ -59,7 +59,7 @@ int main(void)
                 bufp = (char *)calloc(buflen, sizeof(char));
 
                 malloc_trim(0);
-                bytes = klogctl(SYSLOG_ACTION_READ, bufp, log_size);
+                bytes = klogctl(SYSLOG_ACTION_READ, bufp, (int)buflen);
```

Comment:
if (loopcount++ > 0 && buflen > MAX_BUF)

Will be skipped on the first iteration because loopcount = 0 in the beginning, and won't be incremented until after that statement is evaluated, so your buflen will possibly be > MAX_BUF depending on what SYSLOG_ACTION_SIZE_BUFFER returns. Thus you're guaranteed to have allocated at minimum MAX_BUF size bufp on the first run of this. Maybe you want ++loopcount? Or just always use what SYSLOG_ACTION_SIZE_BUFFER returns?